src/components/VeggyBody/ProductList.jsx

```diff
@@ -0,0 +1,102 @@
+import { useState } from "react";
+import ProductItem from "./VeggyProductItem/ProductItem.jsx";
+import Modal from "./Modal/Modal.jsx";
+import "./ProductList.css";
+import NoFoundProductImg from "./../../../public/img/no-product-found.png";
+
+export default function ProductList(props) {
+    const {
+        products,
+        searchedProductName,
+        selectedProducts,
+        setSelectedProducts,
+        setShakingCart,
+        shakingCart,
+    } = props;
+
+    const [modalActive, setModalActive] = useState(false);
+    const [imageValue, setImageValue] = useState({ src: null, name: null, price: null });
+    const filterProductsItem = products.filter(product => product.name.toLowerCase().startsWith(searchedProductName));
+
+    function zoomProductImage (e) {
+        setModalActive(modalActive => !modalActive);
+
+        const parentNode = e.target.closest('.productItem');
+        let getImageProduct = e.target.src;
+        let getProductId = parentNode.id;
+        let getProductInfo = products.find(product =>
+            product.id === parseInt(getProductId)
+        );
+
+        setImageValue({
+            src : getImageProduct,
```

Comment:
Я вродь залишав зауваження в попередньому ПР. 
Навіщо тобі e.target.src виносити в окрему змінну? Зразу пиши 
```suggestion
            src : e.target.src,
```